extended-statistics.md

```diff
@@ -0,0 +1,114 @@
+
+After the TiDB has the extended statistics for correlation, the optimizer can estimate how many rows we need to scan more precisely. Since the `col1` and `col2` are strictly correlated in order, the optimizer will equivalently translate the row count estimate for option two above into:
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM t WHERE col1 <= 1 OR col1 IS NULL;
+```
+
+The above estimation plus one will be the final estimation for the condition. This way, we don't need to use the independent assumption to get a significant estimation error.
+The optimizer will use the independent assumption if the correlation factor is less than the system variable `tidb_opt_correlation_threshold`. But it will increase the estimation heuristically. The larger the system variable `tidb_opt_correlation_exp_factor` is, the larger the estimation result is. The larger the absolute value of the correlation factor is, the larger the estimation result is.
+
+## The collection of the Extended Statistics
+After registration, the TiDB will collect the extended statistic after triggering the ANALYZE command manually or automatically. TiDB will not collect the extended statistics if we only collect the indexes' statistics. It also will not collect it if it's `ANALYZE INCREMENTAL` or the `tidb_enable_fast_analyze` is true.
+
+## The cache of the Extended Statistics
+
+Each TiDB node will maintain a cache for the extended statistics to improve the efficiency of visiting the extended statistics. TiDB will load the table `mysql.stats_extended` periodically to ensure that the cache is kept the same as the data in the table. Each row in the table `mysql.stats_extended` records a column `version`. Once the row is updated, the value of the column `version` will be increased so that we can load the table into the memory incrementally instead of a full loading.
+To delete a record of the extended statistics, TiDB provides the following command:
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE table_name DROP STATS_EXTENDED stats_name;
+```
+
+This command will mark the value of the corresponding record in the table `mysql.stats_extended`'s column `status` to `2`(meaning that the record is deleted) instead of deleting the record directly. Other TiDBs will read this change and delete the record in their memory cache. The background garbage collection will delete the record eventually.
+
+Don't operate the table `mysql.stats_extended` directly. This can cause the inconsistency of the cache of each TiDB node. If you do such an operation wrongly, you can use the following command to load the data of the table fully instead of incrementally:
+
+{{< copyable "sql" >}}
+
+```sql
+ADMIN RELOAD STATS_EXTENDED;
+```
+
+## The dump and load of the Extended Statistics
+
+The way mentioned in the chapter [Introduction to Statistics](/statistics.md) is also suitable for extended statistics. The dump result is in the same JSON file as the normal statistics.
+
+## The switch
```

Comment:
Since extended statistics is disabled by default, it makes more sense to put "how to enable extended statistics" to the beginning of this page.

Reply:
+1